src/sankeydiagram.ts

```diff
@@ -420,20 +475,46 @@ export class SankeyDiagram implements IVisual {
             dataView.categorical.categories[SankeyDiagram.DestinationCategoryIndex].source.displayName,
             dataView.categorical.values ? dataView.categorical.values[SankeyDiagram.FirstValueIndex].source.displayName : null
         );
-
         let cycles: SankeyDiagramCycleDictionary = this.checkCycles(nodes);
-
         if (settings.cyclesLinks.drawCycles === CyclesDrawType.Duplicate) {
             links = this.processCyclesForwardLinks(cycles, nodes, links, settings);
         }
-
-        let sankeyDiagramDataView = {
+        const savedNodes = new Set<SankeyDiagramNode>();
+        settings._linkPositions.forEach(l => { savedNodes.add(l.source); savedNodes.add(l.destination); });
+        let savedNodesArray: SankeyDiagramNode[] = [];
+        savedNodes.forEach(node => savedNodesArray.push(node));
+        let sankeyDiagramDataView: any = {
             nodes,
             links,
             settings,
             columns: []
         };
-
+        settings.filterDetected = savedNodesArray.length !== nodes.length;
+        if (settings.filterDetected) {
+            settings._linkPositions.forEach(l => l.display = true);
+            if (settings.general.pinNodes) {
+                settings._linkPositions.forEach(link => {
+                    link.identity = this.visualHost.createSelectionIdBuilder().createSelectionId();
+                    link.display = false;
+                });
+                links.forEach(link => {
+                    let foundLink: SankeyDiagramLink = settings._linkPositions.find(slink => slink.source.label.name === link.source.label.name && slink.destination.label.name === link.destination.label.name);
```

Comment:
replace the following condition by calling one of two possible methods:
- to compare two values: **source** and **destinations**, and cover it by unit-tests.
- to search all links by your link.

slink.source.label.name === link.source.label.name && slink.destination.label.name === link.destination.label.name